lib/minitest/reporters/html_reporter.rb

```diff
@@ -0,0 +1,212 @@
+require 'builder'
+require 'fileutils'
+require 'erb'
+
+module Minitest
+  module Reporters
+    # A reporter for generating HTML test reports
+    # This is recommended to be used with a CI server, where the report is kept as an artifact and is accessible via a shared link
+    #
+    # The reporter sorts the results alphabetically and then by results so that failing and skipped tests are at the top.
+    #
+    # When using Minitest Specs, the number prefix is dropped from the name of the test so that it reads well
+    #
+    # On each test run all files in the reports directory are deleted, this prevents a build up of old reports
+    #
+    # The report is generated using using ERB. A custom ERB template can be provided but it is not required
+    # The default ERB template uses JQuery and Boostrap, both of these are included by referencing the CDN sites
+    class HtmlReporter < BaseReporter
+
+      # The title of the report
+      attr_reader :title
+
+      # The number of tests that passed
+      def passes
+        count - failures - errors - skips
+      end
+
+      # The percentage of tests that passed, calculated in a way that avoids rounding errors
+      def percent_passes
+        100 - percent_skipps - percent_errors_failures
+      end
+
+      # The percentage of tests that were skipped
+      def percent_skipps
+        (skips/count.to_f * 100).to_i
+      end
+
+      # The percentage of tests that failed
+      def percent_errors_failures
+        ((errors+failures)/count.to_f * 100).to_i
+      end
+
+      # Trims off the number prefix on test names when using Minitest Specs
+      def friendly_name(test)
+        groups = test.name.scan(/(test_\d+_)(.*)/i)
+        return test.name if groups.empty?
+        "it #{groups[0][1]}"
+      end
+
+      # The constructor takes a hash, and uses the following keys:
+      # :title - the title that will be used in the report, defaults to 'Test Results'
+      # :reports_dir - the directory the reports should be written to, defaults to 'test/html_reports'
+      # :erb_template - the path to a custom ERB template, defaults to the supplied ERB template
+      # :mode - Useful for debugging, :terse suppresses errors and is the default, :verbose lets errors bubble up
+      def initialize(args = {})
+        super({})
+
+        defaults = {
+            :title    => 'Test Results',
```

Comment:
why do you use several spaces here and on the next line, but just one space on two others?